module/zfs/ddt.c

```diff
@@ -503,11 +503,14 @@ ddt_get_dedup_histogram(spa_t *spa, ddt_histogram_t *ddh)
 {
 	for (enum zio_checksum c = 0; c < ZIO_CHECKSUM_FUNCTIONS; c++) {
 		ddt_t *ddt = spa->spa_ddt[c];
-		for (enum ddt_type type = 0; type < DDT_TYPES; type++) {
-			for (enum ddt_class class = 0; class < DDT_CLASSES;
-			    class++) {
-				ddt_histogram_add(ddh,
-				    &ddt->ddt_histogram_cache[type][class]);
+		if (ddt) {
```

Comment:
To answer why this is at all relevant - if this is called early in pool import (which the spa_update_dspace() call I added will do), this condition is false, which leads to "fun" a few functions deeper when it actually dereferences.